pkg/R/DataFrame.R

```diff
@@ -774,21 +774,85 @@ setMethod("foreachPartition",
           })
 
 
-############################## DSL ##################################
+############################## SELECT ##################################
+
+getColumn <- function(x, c) {
+  column(callJMethod(x@sdf, "col", c))
+}
 
 setMethod("$", signature(x = "DataFrame"),
           function(x, name) {
-            column(callJMethod(x@sdf, "col", name))
+            getColumn(x, name)
+          })
+
+setMethod("$<-", signature(x = "DataFrame"),
+          function(x, name, value) {
+            stopifnot(class(value) == "Column")
```

Comment:
Does `value` need to be limited to `Column` exclusively? I know that we'd want to pass a Column expression in the scenario where we're mutating a column, but I could also see people wanting to use this as a shortcut for `alias` in which case it would nice to be able to pass in a string for `value`.